### PR TITLE
Simplify the examples that are troublesome for the transformer

### PR DESCRIPTION
**Pull request by warpech**
### Context
This PR simplifies (or makes more consistent) some of the examples that were troublesome for the transformer. Specifically, what has changed:

1. Remove the use of `Handsontable.dom.addEvent` in favor of `addEventListener`
2. Make sure that all inputs (incl. buttons) are in `<div class="controls">`
    - which is located below or above the table. I will discuss this with @jansiegel 
3. Make sure that the outputs are always in `<output>` and filled in using `innerText`
    - especially, on page `batch-operations.md` make sure that the function that uses `innerText` does not rely on `querySelector` and `createElement`.
4. Remove the use of `<label for=""><input id="">`. Now we only use `<label><input></label>`

Pages that were changed:
- http://localhost:8080/docs/next/javascript-data-grid/searching-values/ (Examples 1-4)
- http://localhost:8080/docs/next/javascript-data-grid/batch-operations/ (Live demo)
- http://localhost:8080/docs/next/javascript-data-grid/basic-clipboard/ (Example 2)
- http://localhost:8080/docs/next/javascript-data-grid/selection/ (Examples 1-3)
- http://localhost:8080/docs/next/javascript-data-grid/cell-renderer/ (Examples 1-2)
- http://localhost:8080/docs/next/javascript-data-grid/cell-validator/ (Example 1)
- http://localhost:8080/docs/next/javascript-data-grid/formula-calculation/ (Named expressions) 
- http://localhost:8080/docs/next/javascript-data-grid/searching-values/ (Examples 1-4)
- http://localhost:8080/docs/next/javascript-data-grid/cell-editor/ (only text)
- http://localhost:8080/docs/next/javascript-data-grid/setting-options/ (only cleanup)
- http://localhost:8080/docs/next/react-data-grid/react-hot-column/ (Using the renderer/editor components within React's Context)
- http://localhost:8080/docs/next/react-data-grid/react-hot-reference/ (Example 1)
- http://localhost:8080/docs/next/react-data-grid/react-simple-example/ (External control)
- http://localhost:8080/docs/next/javascript-data-grid/vue-hot-column/ (only cleanup)
- http://localhost:8080/docs/next/javascript-data-grid/saving-data/ ("Load" and "Save" buttons don't work in Docker for reasons that seem unrelated to this PR)

Pages partially fixed:
- http://localhost:8080/docs/next/javascript-data-grid/column-filter/ (very complex examples that need more manual work)
- http://localhost:8080/docs/next/javascript-data-grid/events-and-hooks (very complex example that needs more manual work)
- http://localhost:8080/docs/next/javascript-data-grid/grid-size/ (too complex example that needs simplification)

### How has this been tested?

Locally, using `npm run docs:build`.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. #9592
2.
3.

### Affected project(s):
- [ ] `handsontable`
- [ ] `@handsontable/angular`
- [ ] `@handsontable/react`
- [ ] `@handsontable/vue`
- [ ] `@handsontable/vue3`

### Checklist:

- [ ] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [ ] I have signed the [Contributor License Agreement](https://docs.google.com/forms/d/e/1FAIpQLScpMq4swMelvw3-onxC8Jl29m0fVp5hpf7d1yQVklqVjGjWGA/viewform?c=0&w=1)
- [ ] My change requires a change to the documentation.

[skip changelog]

### CODE REVIEW

**Review by wszymanski — CHANGES_REQUESTED** [review]
Just minor remarks. It looks good.

Btw. It's not related to the task probably. I can see that images on http://localhost:8080/docs/next/javascript-data-grid/cell-renderer/#rendering-custom-html-in-cells page aren't seen.

File: docs/next/guides/advanced-topics/batch-operations.md [hunk 1]
```diff
@@ -166,14 +166,17 @@ The following examples show how much the [`batch()`](@/api/core.md#batch) method
...
-</p>
-<div id="logOutput"></div>
+</div>
+<output class="console" id="output">Here you will see the log</output>
 ```
 ```js
 const container = document.querySelector('#example1');
+const buttonWithout = document.querySelector('#buttonWithout');
+const buttonWith = document.querySelector('#buttonWith');
+const console = document.querySelector('#output');
```
[the remark this answers is not in the document]
Comment (warpech) [reply docs/next/guides/advanced-topics/batch-operations.md]: Fixed in f1c037b, db8d255

**Comment by warpech** [comment]
Ready for re-review

**Review by wszymanski — APPROVED** [review]

**Comment by warpech** [comment]
> Pages partially fixed:
> 
> - http://localhost:8080/docs/next/javascript-data-grid/column-filter/ (very complex examples that need more manual work)
> - http://localhost:8080/docs/next/javascript-data-grid/events-and-hooks (very complex example that needs more manual work)
> - http://localhost:8080/docs/next/javascript-data-grid/grid-size/ (too complex example that needs simplification)

This will be addressed later, maybe in PR #9623

**Pull request merged**